### DISCUSSION TITLE
Add a circle background to the image using libvips command line

### DISCUSSION THREAD

**Discussion by WeisiminPeng-Simmi** [libvips/libvips#3571] [Q&A] [answered]
Hi I am new to libvips, I want to implement what I used to do using ImageMagick. Create a circle background for my image using command line. It will be awesome if libvips has even easy way to do it. Could you please help me? Thanks!

The command I used to use in ImageMagick
```
// create circle background
convert -size 1200x1200 xc:none -fill grey -gravity center -draw "circle 600,600 1198,600" background.png
// output image
composite -gravity center -compose over -colorspace sRGB input.png background.png output.png
```

![b1](https://github.com/libvips/libvips/assets/139998239/771c252f-4a94-479a-ba78-80404467c5fd)
![dog](https://github.com/libvips/libvips/assets/139998239/b4d0d9e8-5b01-4561-acf1-6957853d5768)
![output](https://github.com/libvips/libvips/assets/139998239/4fef03e2-8c77-499d-a2dc-86c59575d4a7)

**Comment by jcupitt** [comment, marked as answer]
Hello @WeisiminPeng-Simmi,

I'd use python for something like this. Maybe:

```python
#!/usr/bin/python3

import sys
import pyvips

image = pyvips.Image.new_from_file(sys.argv[1], access="sequential")

# make something about 50% bigger
width = image.width * 1.5
height = image.height * 1.5

centre_x = width / 2
centre_y = height / 2
r = min(centre_x, centre_y)

background = pyvips.Image.svgload_buffer(bytes(f"""
<svg viewBox="0 0 {width} {height}">
    <circle cx="{centre_x}" cy="{centre_y}" r="{r}" fill="#bebebe" />
</svg>
""", "utf8"))

image = background.composite2(image, "over", 
                              x=(width - image.width) / 2,
                              y=(height - image.height) / 2)

image.write_to_file(sys.argv[2])
```

You can run it to put any image on a grey circle background, eg.:

```
$ ./circle-background.py ~/pics/dog.png x.png
```

To make:

![x](https://github.com/libvips/libvips/assets/580843/c7e46cae-cd0d-4ca6-a6e1-0a1c9812c39c)

**Reply by jcupitt** [reply]
Ah you said on the command line. The libvips CLI is deliberately pretty basic, you'd be better off with a proper scripting language for anything complex. libvips is available for most languages, so python, ruby, php, lua, etc.

**Reply by jcupitt** [reply]
... the advantage of a proper scripting language is much greater speed and lower memory use. I tried your imagemagick version vs. the pyvips one:

```
$ /usr/bin/time -f %M:%e ./circle-background.sh ~/pics/dog.png x.png
38644:0.37
$ /usr/bin/time -f %M:%e ./circle-background.py ~/pics/dog.png x.png
69588:0.12
```

So pyvips is 3x faster and needs 5x less memory.
